### PR TITLE
Fix for some Visual C++ x64 compile warnings

### PR DESCRIPTION
**Pull request by pollocktoci**
Spdp.cpp - fix for "time_t to CORBA::Long" conversion warning
DataReaderListener.cpp - fix for "size_t to const int" conversion warning.
Domain.h - fix for "size_t to CORBA::ULong" conversion warning.
ut_BIT_DataReader.cpp - fix for "long to CORBA::Octet" conversion warning.

### CODE REVIEW

File: tests/DCPS/LargeSample/DataReaderListener.cpp [hunk 1]
```diff
@@ -215,8 +215,12 @@ bool DataReaderListenerImpl::data_consistent() const
                   << writer->second.size() << std::endl;
         valid_and_done = false;
         if (writer->second.size() < samples_per_writer_) {
-          for (size_t sample = 0; sample < samples_per_writer_; ++sample) {
-            std::cerr
+#if __cplusplus >= 199707L
```
Comment (mitza-oci) [inline tests/DCPS/LargeSample/DataReaderListener.cpp:218]: Why is this conditional?  Isn't `samples_per_writer_` always a `size_t`?

Comment (pollocktoci) [reply tests/DCPS/LargeSample/DataReaderListener.cpp:218]: Oh, I think you're right.  I'm at a disadvantage when trying to fix warnings in that my Visual Studio install stupidly doesn't reproduce them.  It's the passing of 'sample' to 'writer->second.count(sample)', and to 'expected_data_field_length' that is producing the warnings.  'count' wants a const int.  'expected_data_field_length' wants an int.

Comment (pollocktoci) [reply tests/DCPS/LargeSample/DataReaderListener.cpp]: I modified this to cast the size_t to the types being complained about.  I was able to get these warnings to show up in my visual studio by changing the DDS_LargeSample_Subscriber project settings to show all warnings.  I'm now seeing these warnings, which this change fixed, along with lots of other warnings.

File: tests/DCPS/Observer/Domain.h [hunk 2]
```diff
@@ -11,8 +11,13 @@ class Domain {
 
   template<typename Qos>
   static void change_qos(Qos& qos, const std::string& data) {
+#if __cplusplus >= 199707L
```
Comment (mitza-oci) [inline tests/DCPS/Observer/Domain.h:14]: Similar to the note above: I don't know of anything different based on the C++ versions.  The Standard Library types always use `size_t` and the IDL Sequences (C++ classic mapping) use a 32-bit unsigned integer.

File: tests/DCPS/LargeSample/DataReaderListener.cpp [hunk 3]
```diff
@@ -215,14 +215,14 @@ bool DataReaderListenerImpl::data_consistent() const
                   << writer->second.size() << std::endl;
         valid_and_done = false;
         if (writer->second.size() < samples_per_writer_) {
-          for (size_t sample = 0; sample < samples_per_writer_; ++sample) {
-            std::cerr
-              << (writer->second.count(sample) ? "           Got" : "ERROR: Missing")
+            for (size_t sample = 0; sample < samples_per_writer_; ++sample) {
```
Comment (mitza-oci) [inline tests/DCPS/LargeSample/DataReaderListener.cpp:218]: Indent level changed here

**Review by mitza-oci — CHANGES_REQUESTED** [review]

**Review by mitza-oci — APPROVED** [review]

**Comment by mitza-oci** [comment]
These warnings show up on the Azure CI build:
```
D:\a\1\s\dds\idl\langmap_generator.cpp(237,31): warning C4018: '<': signed/unsigned mismatch [D:\a\1\s\dds\idl\opendds_idl.vcxproj]
D:\a\1\s\tests\DCPS\Observer\Domain.h(14,1): warning C4267: 'argument': conversion from 'size_t' to 'CORBA::ULong', possible loss of data [D:\a\1\s\tests\DCPS\Observer\Observer_Pub.vcxproj]
D:\a\1\s\tests\DCPS\Observer\Domain.h(16,1): warning C4267: 'argument': conversion from 'size_t' to 'CORBA::ULong', possible loss of data [D:\a\1\s\tests\DCPS\Observer\Observer_Pub.vcxproj]
```

**Pull request merged**